### PR TITLE
Update the list of functions in expression

### LINKED ISSUE #684: [FEATURE] New 'reverse' expression for reversing linestrings
Original commit: https://github.com/qgis/QGIS/commit/da436f926be5876245067dc1c8d22c0bfefcca5e by nyalldawson

Unfortunately this lazy coder did not write a description... :-(

### LINKED ISSUE #701: [FEATURE] Add an eval expression
Original commit: https://github.com/qgis/QGIS/commit/f961ecef465fd0dede546670356dc373b7a36ec5 by m-kuhn

Funded by
 * Regional Council of Picardy
 * ADUGA
 * Ville de Nyon
 * Wetu GIT cc

### LINKED ISSUE #702: [FEATURE] Add translate expression function
Original commit: https://github.com/qgis/QGIS/commit/4b3083d9a7163bdd19785289e6597b51b396a71f by m-kuhn

Funded by
 * Regional Council of Picardy
 * ADUGA
 * Ville de Nyon
 * Wetu GIT cc

### LINKED ISSUE #707: [FEATURE] add darker() and lighter() expression functions
Original commit: https://github.com/qgis/QGIS/commit/a5d670233baac952996a71ccb96c46b5bb781c79 by nyalldawson

Unfortunately this lazy coder did not write a description... :-(

### LINKED ISSUE #716: [FEATURE] exterior_ring function for expressions
Original commit: https://github.com/qgis/QGIS/commit/262514d97d0ac754e5c2989a2ae85c0eb54c7242 by nyalldawson

Returns exterior ring of a polygon as a line. Useful for geometry
generator symbology for styling only the exterior ring of
polygons.

### LINKED ISSUE #719: [FEATURE] is_closed function for expressions
Original commit: https://github.com/qgis/QGIS/commit/ceddb7eb97c42749ce5f1b28e7702f9fca111061 by nyalldawson

Returns whether a linestring is closed

### LINKED ISSUE #725: [FEATURE] More geometry functions for expressions
Original commit: https://github.com/qgis/QGIS/commit/5b244ae88ce2c0d851f52a3aaef971bd5beb10c7 by nyalldawson

- add accessors: geometry_n, interior_ring_n
- add num_geometries, num_rings, num_interior_rings
- add nodes_to_points for converting every node in a geometry
to a multipoint geometry
- add segments_to_lines for converting every segment in a geometry
to a multiline geometry

nodes_to_points and segments_to_lines are intended for use with
geometry generator symbology, eg to allow use of m and z values
for nodes/lines with data defined symbology.

### LINKED ISSUE #755: [FEATURE] Add closest_point and shortest_line expression functions
Original commit: https://github.com/qgis/QGIS/commit/275eb9463d5badea928710313b3ec878a14310db by nyalldawson

closest_point: returns closest point a geometry to a second geometry
shortest_line: returns the shortest possible line joining two
geometries

### LINKED ISSUE #757: [FEATURE] Expression variables relating to QGIS environment
Original commit: https://github.com/qgis/QGIS/commit/91f3005f806a75308dd4726272c2ac5f113858db by nyalldawson

New variables for:
- @qgis_os_name: eg 'linux','windows' or 'osx'
- @qgis_platform: eg 'desktop' or 'server'
- @user_account_name: current user's operating system account name
- @user_full_name: current user's name from os account (if available)

Sponsored by Andreas Neumann

### PR DESCRIPTION
**Pull request by DelazJ**
Adds new functions : fixes #757, fixes #755, fixes #725, fixes #719, fixes #716, fixes #707, fixes #702 , fixes #701, fixes #684

Previously, functions were listed inside a quote and thus, were untranslatable.
Updating the list was also complicated given that a quote item doesn't well handle words;
they could be wrapped anywhere leading to an unreadable description.
Functions are now shown inside a table (which better handles multiline and word wrap --> easy to update). Their description also become translatable.

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/expression.rst [hunk 1]
```diff
@@ -251,56 +269,57 @@ Date and Time Functions
...
- year           extracts the year part from a date, or the number of  
+ year           Extracts the year part from a date or datetime, or the number of 
                 years from an interval
-
+==============  ===================================================================
 
 **Some example:**
 
-* Get the month and the year of today in the format "10/2014" ::
+* Get the month and the year of today in the format "02/2016" ::
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/expression.rst]: is really worth to update the date? This lead to update the translation for no added value.

**Comment by yjacolin** [comment]
huge work to review :) could you give me your opinion about my comments? Just good to merge for me.

Note: next time, could you split your PR (not two commits) in two PRs: one for the formating change and another for new text? It will be make easier the review :)

**Comment by DelazJ** [comment]
@yjacolin I noted your advice. I will try to follow it next time although it might not be that easy to separate matters at the beginning.
sorry for the potential trouble for review. :(

**Comment by yjacolin** [comment]
I know, I have the same problem in one of my PR :) Just add a comment in a commit (eg "No documentation change") et we can review commit by commit (look much simpler this way)

**Pull request merged**